### PR TITLE
feat(docker): setup frontend docker container

### PR DESCRIPTION
**Pull request by JohnL17**

### CODE REVIEW

File: caluma-demo/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  location = /index.html {
+      expires epoch;
+      add_header Cache-Control 'private, no-store, must-revalidate, max-age=0';
+  }
+
+  location = /sw.js {
+      expires epoch;
+      add_header Cache-Control 'private, no-store, must-revalidate, max-age=0';
+  }
```
Comment (anehx) [inline caluma-demo/nginx.conf]: I think we are not using a service worker - this is unnecessary...

File: caluma-demo/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  location = /manifest.webmanifest {
+      expires 30d;
+      add_header Cache-Control 'public, max-age=2592000';
+  }
```
Comment (anehx) [inline caluma-demo/nginx.conf]: Same here, we don't have a webmanifest

**Review by anehx — CHANGES_REQUESTED** [review]

File: caluma-demo/nginx.conf (hunk 1 above)
Comment (JohnL17) [reply caluma-demo/nginx.conf]: I have deleted these lines.

File: caluma-demo/nginx.conf (hunk 2 above)
Comment (JohnL17) [reply caluma-demo/nginx.conf]: I have deleted these lines too.

File: caluma-demo/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    try_files $uri $uri/ /index.html?/$request_uri;
+  }
+
+  location ~ ^/graphql {
+    proxy_pass http://caluma:8000;
+  }
+
+  location ~ ^/static/ {
+      proxy_pass http://caluma:8000;
+  }
```
Comment (anehx) [inline caluma-demo/nginx.conf]: Remove this

**Review by anehx — CHANGES_REQUESTED** [review]

File: caluma-demo/nginx.conf (hunk 3 above)
Comment (JohnL17) [reply caluma-demo/nginx.conf]: The unnecessary line is removed

**Review by anehx — APPROVED** [review]

File: caluma-demo/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  gzip_static on;
+  gzip_proxied any;
+
+  location / {
+    add_header Cache-Control 'private, max-age=86400';
+    try_files $uri $uri/ /index.html?/$request_uri;
+  }
+
+  location ~ ^/graphql {
```
Comment (kaldras) [inline caluma-demo/nginx.conf]: In our other projects we add a trailing slash to the location regex. I think this makes sense to make sure the intended path (in this case `/graphql/`) is matched and not a random one (for example `/graphqli/`).

File: docker-compose.yml [hunk 5]
```diff
@@ -4,51 +4,34 @@ services:
...
-      # https://hub.docker.com/_/postgres
-      # - POSTGRES_PASSWORD=
     volumes:
       - dbdata:/var/lib/postgresql/data
     command: -p ${DATABASE_PORT:-5432}
   caluma:
     image: projectcaluma/caluma:latest
     ports:
-      - "8000:8000"
+      - "8000"
```
Comment (kaldras) [inline docker-compose.yml]: Does this even make sense if no port mapping is done?

**Review by kaldras — CHANGES_REQUESTED** [review]

File: caluma-demo/nginx.conf (hunk 4 above)
Comment (JohnL17) [reply caluma-demo/nginx.conf]: I wrote in the environment.js file "/graphql". That was the reason why I didn't add a slash in the end in the nginx.conf. Now it's fixed.

File: docker-compose.yml (hunk 5 above)
Comment (JohnL17) [reply docker-compose.yml]: No it doesn't make sense. I have removed it.

File: caluma-demo/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+WORKDIR /app
+
+RUN yarn
+
+COPY . /app
+
+RUN yarn ember deploy production
+
+FROM nginx:alpine as final
```
Comment (kaldras) [inline caluma-demo/Dockerfile]: A second `FROM` in the Dockerfile seems wrong. Where is this coming from? What is the intention?

**Review by kaldras — CHANGES_REQUESTED** [review]

File: caluma-demo/Dockerfile (hunk 6 above)
Comment (kaldras) [reply caluma-demo/Dockerfile]: Ok, as discussed.

File: caluma-demo/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+COPY --from=0 /app/tmp/deploy-dist /usr/share/nginx/html
```
Comment (kaldras) [inline caluma-demo/Dockerfile]: Why do we name the last stage (but don't seem to use it) and don't name the first stage which we use here? Would be nice to do it consistently.

Comment (JohnL17) [reply caluma-demo/Dockerfile]: Ah right. I wanted to rename the stages and forgot it. Now they are renamed.

**Review by kaldras — APPROVED** [review]

**Pull request merged**